jep/202/README.adoc

```diff
@@ -0,0 +1,230 @@
+= JEP-202: External Artifact Storage
+:toc: preamble
+:toclevels: 3
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+.Metadata
+[cols="2"]
+|===
+| JEP
+| 202
+
+| Title
+| External Artifact Storage
+
+| Sponsor
+| https://github.com/carlossg[@carlossg]
+
+// Use the script `set-jep-status <jep-number> <status>` to update the status.
+| Status
+| Draft :speech_balloon:
+
+| Type
+| Standards
+
+| Created
+| 2018 Apr 13
+//
+//
+// Uncomment if there is an associated placeholder JIRA issue.
+//| JIRA
+//| :bulb: https://issues.jenkins-ci.org/browse/JENKINS-nnnnn[JENKINS-nnnnn] :bulb:
+//
+//
+// Uncomment if there will be a BDFL delegate for this JEP.
+//| BDFL-Delegate
+//| :bulb: Link to github user page :bulb:
+//
+//
+// Uncomment if discussion will occur in forum other than jenkinsci-dev@ mailing list.
+//| Discussions-To
+//| :bulb: Link to where discussion and final status announcement will occur :bulb:
+//
+//
+// Uncomment if this JEP depends on one or more other JEPs.
+//| Requires
+//| :bulb: JEP-NUMBER, JEP-NUMBER... :bulb:
+//
+//
+// Uncomment and fill if this JEP is rendered obsolete by a later JEP
+//| Superseded-By
+//| :bulb: JEP-NUMBER :bulb:
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+
+== Abstract
+
+Jenkins uses the master filesystem to store all generated artifacts, unless explicitly using a plugin that archives somewhere else.
+There are serious drawbacks to use filesystem when running in cloud or containerized environments.
+This proposal provides the APIs needed to store all artifacts in an external location without going through the master.
+It includes an initial implementation using AWS S3.
+
+== Specification
+
+Jenkins agents upload and download artifacts directly to the external location using HTTP, no content traffic is sent over the remoting channel.
+
+All external store operations except upload/download are executed from the master, so the agent does not need any permissions other than HTTP upload/download. Pre-signed urls with expiration can be used so the agent does not have access to the full store.
+
+Stashes are stored as tarballs to ensure permissions and links are preserved.
```

Comment:
is it implicit that we use `tar` only, or compressed tar (be it `.tar.gz`, `.tar.xz`...)?

If we do not want to compress it, because for instance we tested it and it's often unneeded, or even counter-productive, I would expect a part explaining this in the _Reasoning_ section below.

Reply:
tgz currently https://github.com/jenkinsci/artifact-manager-s3-plugin/blob/master/src/main/java/io/jenkins/plugins/artifact_manager_s3/JCloudsArtifactManager.java#L212

Reply:
The default `stash` implementation using the master filesystem also uses `tgz`, but yes it is an implementation detail.